SocialCareCaseViewerApi.Tests/V1/UseCase/FormSubmissionUseCaseTests.cs

```diff
@@ -1174,5 +1174,80 @@ public void ExcecuteDeleteCallsMongoDbGaetwayWhenTheSubmissionTypeIsSupported(st
 
             _mockMongoGateway.Verify(x => x.UpsertRecord(CollectionName, ObjectId.Parse(submission.SubmissionId.ToString()), submission), Times.Once);
         }
+
+        [Test]
+        public void SanitiseResidentReturnsResidentWithoutRelatedObjects()
+        {
+            var person = TestHelpers.CreatePersonWithRelatedAttributes();
+
+            var address = TestHelpers.CreateAddress(person.Id);
+            address.Person = person;
+            var phoneNumber = TestHelpers.CreatePhoneNumber(person.Id);
+            phoneNumber.Person = person;
+            var otherName = TestHelpers.CreatePersonOtherName(person.Id);
+            otherName.Person = person;
+            var allocation = TestHelpers.CreatePersonAllocationSet(person.Id);
+            allocation.Person = person;
+            var warningNote = TestHelpers.CreateWarningNote(person.Id);
+            warningNote.Person = person;
+            var keyContact = TestHelpers.CreateKeyContact(person.Id);
+            keyContact.Person = person;
+            var gpDetails = TestHelpers.CreateGpDetails(person.Id);
+            gpDetails.Person = person;
+            var techUse = TestHelpers.CreateTechUse(person.Id);
+            techUse.Person = person;
+            var disability = TestHelpers.CreateDisability(person.Id);
+            disability.Person = person;
+            var emailAddress = TestHelpers.CreateEmailAddress(person.Id);
+            emailAddress.Person = person;
+            var lastUpdated = TestHelpers.CreateLastUpdated(person.Id);
+            lastUpdated.Person = person;
+
+            person.Addresses.Add(address);
+            person.PhoneNumbers.Add(phoneNumber);
+            person.OtherNames.Add(otherName);
+            person.Allocations.Add(allocation);
+            person.WarningNotes.Add(warningNote);
+            person.KeyContacts?.Add(keyContact);
+            person.GpDetails?.Add(gpDetails);
+            person.TechUse?.Add(techUse);
+            person.Disability?.Add(disability);
+            person.Emails?.Add(emailAddress);
+            person.LastUpdated?.Add(lastUpdated);
+
+            var residents = new List<Person> { person };
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            var workers = new List<Worker> { TestHelpers.CreateWorker() };
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(workers[0]);
+            _mockDatabaseGateway.Setup(x => x.GetPersonDetailsById(request.ResidentId)).Returns(residents[0]);
+            _mockMongoGateway.Setup(x => x.InsertRecord(It.IsAny<string>(), It.IsAny<CaseSubmission>()));
```

Comment:
Because the circular reference issue occurs at this stage I think it would be better to test the shape of the object passed to the`InsertRecord` method rather than testing what comes back from the use case. This way you would be confident that the circular references have been removed from the object to be inserted. The use case has a lot of logic that alters the shape of the object before it's returned, so targeting the input object would be safest option I think.

Reply:
Refactored to check which objects is being inserted